src/custom_language_parsers.py

```diff
@@ -1,60 +1,44 @@
 from collections import defaultdict
 from dataclasses import dataclass, field
+from typing import Optional
 
 from tree_sitter_languages import get_language, get_parser
 
 
+@dataclass
 class ImportedModule:
-    def __init__(
-        self, module_base_name: str, alias: str = None, imported_objects: list = None
-    ) -> None:
-        self.module_base_name = module_base_name
-        self.alias = alias
-        self.imported_objects = imported_objects
-
-    @classmethod
-    def from_tree_node(cls, tree_node):
-        module_base_name = None
-        alias = None
-        imported_objects = None
-        if tree_node.type == 'import_from_statement':
-            module_base_name = tree_node.named_children[0].text.decode('ascii')
-            imported_objects = [
-                ch.text.decode('ascii') for ch in tree_node.named_children[1:]
-            ]
-
-        elif tree_node.type == 'import_statement':
-            for child in tree_node.children:
-                if child.type == 'dotted_name':
-                    module_base_name = child.text.decode('ascii')
-                elif child.type == 'aliased_import':
-                    module_base_name = child.named_children[0].text.decode('ascii')
-                    alias = child.named_children[1].text.decode('ascii')
-        return cls(module_base_name, alias, imported_objects)
+    module_base_name: str
+    alias: str = None
+    imported_objects: list = field(default_factory=list)
 
     def __str__(self) -> str:
         if self.alias is None:
             return self.module_base_name
         return f'{self.module_base_name} as {self.alias}'
 
 
-def match_function_call_to_imported_module(function_call, imported_modules):
+def get_imported_module_for_function_call(
+    function_call: str, imported_modules: list[ImportedModule]
+) -> Optional[ImportedModule]:
     for imported_module in imported_modules:
-        if function_call in imported_module.module_base_name.split('.'):
-            return imported_module
-        if imported_module.imported_objects is not None:
-            for imported_object in imported_module.imported_objects:
-                if function_call == imported_object:
-                    return imported_module
+        if imported_module.imported_objects is None:
+            continue
+        for imported_object in imported_module.imported_objects:
+            if function_call == imported_object:
+                return imported_module
     return None
 
 
 @dataclass
 class CustomLanguageSyntaxParser:
     name: str = 'python'
     extension: str = 'py'
-    parser: object = get_parser('python')
-    language: object = get_language('python')
+    parser: object = Optional[object]
```

Comment:
Nice job on the typing, but object type in python doesn't carry info (since everything is an object (even functions -_-)) so it's best to avoid it.